### DISCUSSION TITLE
Now building both SDL1 and SDL2 versions

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#190] [Announcements]
I implemented SDL2. Had lot of distractions, so barely tested it - please report if you see a difference between how SDL1 and SDL2 versions work.

Within "Actions", The standard version is now built with SDL2, and there's a separate set of files for SDL1 (marked as such in file names).

**Comment by Moburma** [comment]
Yeah, I meant to log this, it's not worked right for me since the start of these changes. I've not tried the commit today, about to go out, but I had to use build 0.3.7.2263 as the last one where menus would load. With no arguments it would hang from the command line on startup for a few seconds, then exit. I could use -m to get into missions in subsequent builds, but changing to low res mode would instantly exit. The menus still would not work It seems it's something to do with the low res screenmodes. 

Example error output:

bflibrary ver 0.1.2.0 (standard release)
LOG CREATED  @ 22:26:00  30-08-2024

Error: Bflib: LbFileOpen: file does not exist: "data/tex06.dat"
Error: Bflib: LbScreenSetupAnyMode: full screen resolution 320x200 (mode 1) not available

**Comment by geist22** [comment]
Doesn't work either for me, the program freezes with a black screen right at the start, regardless of whether I start it normally or try to go directly into a mission from the command line. Latest release, [8f0dd24](https://github.com/swfans/swars/commit/8f0dd24658e0661acd6f9ae8b006c15bee96ab14) as of now.

Error: Bflib: LbFileOpen: file does not exist: "data/tex06.dat"
Error: Bflib: LbPaletteSet: SetPalette to ScreenSurface failed: 
Error: Bflib: LbScreenSetupAnyMode: palette setting failed
Warning: Bflib: LbScreenReset: screen got reset while locked

**Comment by mefistotelis** [comment]
I see. The SDL2 has `SDL_BlitScaled`, so I though this will replace all custom blitting routines we have in bflibrary. I was obviously mistaken - the `SDL_BlitScaled` comes with a lot of undocumented restrictions which in practice make it useless here.

**Comment by Moburma** [comment]
Latest commits seem to work fine for me again, I will test it out to see if I find any issues.

**Comment by mefistotelis** [comment]
I also just remade shooting weapons. So you may want to search for changes in that - the function was quite complex.

**Comment by Moburma** [comment]
I think I found a new bug related to this, only tried it on the latest build. In windows, I'm playing in full screen mode. If I alt+tab out to another program, then come back, the screen is frozen on the last image I saw, and never updates. I can hear the game still running, so it's not paused, but the screen never updates after this point, the game must be quit to play it again.

**Reply by mefistotelis** [reply]
Any errors visible in the log?

**Reply by Moburma** [reply]
Yep, LOTS of:

Error: Bflib: LbScreenSwap: flip failed: Window surface is invalid, please call SDL_GetWindowSurface() to get a new surface

and then once at the end:

Warning: Bflib: LbScreenReset: screen got reset while locked

I accidentally tabbed during the intro as well, that had the same effect there, but once I pressed the mouse button to get to the game proper the screen drawing worked fine for the next part - until I tabbed out again.

Also the SDL2 build doesn't seem to output the commandline information like build number etc.

I tried the latest SDL1 build, but this just hangs on startup like described previously.

**Comment by mefistotelis** [comment]
Both SDL1 and SDL2 should be fixed now.

**Reply by Moburma** [reply]
Yep, they both seem to work fine, screen displays correctly and I can alt+tab both builds without issue.
